app/javascript/src/components/Dashboard/Notes/MenuBar.jsx

```diff
@@ -0,0 +1,79 @@
+import React, { useState } from "react";
+
+import { Search, Settings, Plus } from "neetoicons";
+import { Typography } from "neetoui";
+import { MenuBar } from "neetoui/layouts";
+
+const NotesMenuBar = ({ showMenu }) => {
+  const [isSearchCollapsedSegments, setIsSearchCollapsedSegments] =
+    useState(true);
+  const [isSearchCollapsedTags, setIsSearchCollapsedTags] = useState(true);
```

Comment:
Follow the noun-verb format while naming variables which is more or less a question.

```suggestion
const [isSegmentsSearchCollapsed, ...] =
    useState(true);
  const [isTagsSearchCollapsed, ...] = useState(true);
```

Refer: https://grammar.yourdictionary.com/vs/noun-vs-verb-easy-identification-guide.html